tracer/src/Datadog.Trace/DatabaseMonitoring/DatabaseMonitoringPropagator.cs

```diff
@@ -73,5 +82,74 @@ internal static string PropagateSpanData(DbmPropagationLevel propagationStyle, s
+            {
+                return false;
+            }
+
+            var stopwatch = System.Diagnostics.Stopwatch.StartNew();
+
+            byte version = 0; // version can have a maximum value of 15 in the current format
+            var sampled = SamplingPriorityValues.IsKeep(span.Context.GetOrMakeSamplingDecision() ?? SamplingPriorityValues.Default);
+            var contextValue = BuildContextValue(version, sampled, span.SpanId, span.TraceId128);
+
+            using (var injectionCommand = connection.CreateCommand())
+            {
+                injectionCommand.CommandText = SetContextCommand;
+
+                var parameter = injectionCommand.CreateParameter();
+                parameter.ParameterName = ContextInfoParameterName;
+                parameter.Value = contextValue;
+                parameter.DbType = DbType.Binary;
+                injectionCommand.Parameters.Add(parameter);
+
+                injectionCommand.ExecuteNonQuery();
+
+                if (Log.IsEnabled(LogEventLevel.Debug))
+                {
+                    // avoid building the string representation in the general case where debug is disabled
+                    Log.Debug("Span data for DBM propagated for {Integration} via context_info with value {ContextValue} (propagation level: {PropagationLevel}", integrationId, HexConverter.ToString(contextValue), propagationLevel);
+                }
+            }
+
+            // Since sending the query to the DB can be a bit long, we register the time it took for transparency.
+            // Not using _dd because we want the customers to be able to see that tag.
+            span.SetMetric("dd.instrumentation.time_ms", stopwatch.Elapsed.TotalMilliseconds);
+
+            return true;
+        }
+
+        /// <summary>
+        /// Writes the given info in a byte array with the following format:
+        /// 4 bits: protocol version, 3 bits: reserved, 1 bit: sampling decision, 64 bits: spanID, 128 bits: traceID
+        /// </summary>
+        private static byte[] BuildContextValue(byte version, bool isSampled, ulong spanId, TraceId traceId)
+        {
+            var sampled = isSampled ? 1 : 0;
+            var versionAndSampling = (byte)(((version << 4) & 0b1111_0000) | (sampled & 0b0000_0001));
+            var contextBytes = new byte[1 + sizeof(ulong) + TraceId.Size];
+
+            contextBytes[0] = versionAndSampling;
+            BinaryPrimitives.WriteUInt64BigEndian(
+                new VendoredMicrosoftCode.System.Span<byte>(contextBytes, start: 1, sizeof(ulong)),
+                spanId);
+            BinaryPrimitives.WriteUInt64BigEndian(
+                new VendoredMicrosoftCode.System.Span<byte>(contextBytes, 1 + sizeof(ulong), sizeof(ulong)),
+                traceId.Upper);
+            BinaryPrimitives.WriteUInt64BigEndian(
+                new VendoredMicrosoftCode.System.Span<byte>(contextBytes, 1 + sizeof(ulong) + sizeof(ulong), sizeof(ulong)),
+                traceId.Lower);
```

Comment:
Instead of using `new Span<byte>()` for every call to `WriteUInt64BigEndian()`, you can use `Span.Slice()`. Something like:
```csharp
var span = new VendoredMicrosoftCode.System.Span<byte>(contextBytes)
{
    [0] = versionAndSampling
};

BinaryPrimitives.WriteUInt64BigEndian(span.Slice(1), spanId);
BinaryPrimitives.WriteUInt64BigEndian(span.Slice(1 + sizeof(ulong)), traceId.Upper);
BinaryPrimitives.WriteUInt64BigEndian(span.Slice(1 + sizeof(ulong) + sizeof(ulong)), traceId.Lower);
```

Reply:
ooh ok nice, thanks. Manipulating spans (I mean, those ones) is pretty new to me :)